my diy kata

gmiellis opened the pull request.

joestephens commented on __tests__/kata2.booleanToWord.test.js:
```diff
@@ -2,4 +2,13 @@ const { booleanToWord } = require('../src');
   // how do we create specs again???
+
+it('returns Yes when booleanToWord is True',() => {
+  expect(booleanToWord(true)).toBe('Yes')
```
Always be consistent with your indentation method: either go with 4 spaces or 2. I recommend 2.

joestephens commented on __tests__/kata4.humanCatDogYears.test.js:
```diff
@@ -1,3 +1,9 @@
 const { humanCatDogYears } = require('../src');
 
 // Look Ma, no handlebars!!!
```
You shouldn't leave comments in your git commits unless they are explaining a piece of code that would otherwise be hard to interpret.

joestephens commented on src/kata1.fizzBuzz.js:
```diff
@@ -1,5 +1,17 @@
+    const isDivisableBy3 = (number % 3) === 0;
+    const isDivisableBy5 = (number % 5) === 0;
 
+if (isDivisableBy3 && isDivisableBy5) {
```
INDENTATION!

joestephens commented on src/kata2.booleanToWord.js:
```diff
@@ -1,5 +1,10 @@
+        return 'Yes';
+    } else {
+        return 'No';
+    }
```
You could do this with a [ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) (a shorthand if / else statement):

```js
return boolean ? 'Yes' : 'No'
```

joestephens commented on src/kata3.numberToReversedDigits.js:
```diff
@@ -1,5 +1,12 @@
 
+const tostring = number.toString();
+const split = tostring.split('');
+let map = split.map(function(result) {
```
`map` should be a constant variable here as it isn't reassigned to. I'd also use an arrow function with an implicit return:

```js
const map = split.map(result => parseInt(result));
```

joestephens commented on src/kata4.humanCatDogYears.js:
```diff
@@ -1,5 +1,25 @@
+  let hcdy = [];
+    hcdy.push(number);
+
+    const cat = function (number) {
```
Really good thought process here in breaking the functionality up into different functions. I would probably not be afraid to make these more concise though:

```js
const cat = number => (((number - 2) * 4) + 24)
```

joestephens commented on src/kata4.humanCatDogYears.js:
```diff
@@ -1,5 +1,25 @@
+
+    hcdy.push(dog(number));
+
+    return hcdy;
```
`hcdy` isn't very meaningful a name, and you probably don't need it:

```js
return [human(number), cat(number), dog(number)];
```

joestephens commented on src/kata5.reachDestination.js:
```diff
@@ -1,5 +1,26 @@
 const reachDestination = (distance, speed) => {
 
+  function reachDestination (distance, speed) {
```
Not sure why you have a function within a function here - you definitely don't need it!

joestephens commented on src/kata5.reachDestination.js:
```diff
@@ -1,5 +1,26 @@
+            twoDecArr[1] = 0;
+      };
+
+    let result = twoDecArr.join('.');
```
Use constant variables unless it's your intention to reassign (you should avoid reassignment, but it is acceptable for mathematical operations).

gmiellis commented:
Morning Joe,
Thanks for the feedback. It's good to know I'm on the right track and I will take your comments onboard an hopefully now with the help of ESLint I will get into some good habits.

I somehow overlooked this email and only just saw it this morning.
Thanks again.
Gaz

   On Thursday, 21 June 2018, 19:38:34 GMT+1, Joe Stephens <notifications@github.com> wrote: 
 
 
@joestephens commented on this pull request.

Some really good approaches to some of these problems. Just be careful with indentation and also become more comfortable with using const over let.

In __tests__/kata2.booleanToWord.test.js:
> @@ -2,4 +2,13 @@ const { booleanToWord } = require('../src');
 
 describe('booleanToWord', () => {
   // how do we create specs again???
+
+it('returns Yes when booleanToWord is True',() => {
+  expect(booleanToWord(true)).toBe('Yes')

Always be consistent with your indentation method: either go with 4 spaces or 2. I recommend 2.

In __tests__/kata4.humanCatDogYears.test.js:
> @@ -1,3 +1,9 @@
 const { humanCatDogYears } = require('../src');
 
 // Look Ma, no handlebars!!!

You shouldn't leave comments in your git commits unless they are explaining a piece of code that would otherwise be hard to interpret.

In src/kata1.fizzBuzz.js:
>  
+if (isDivisableBy3 && isDivisableBy5) {

INDENTATION!

In src/kata2.booleanToWord.js:
> @@ -1,5 +1,10 @@
 const booleanToWord = (boolean) => {
-
-}
+    
+    if (boolean) {
+        return 'Yes';
+    } else {
+        return 'No';
+    }

You could do this with a ternary (a shorthand if / else statement):
return boolean ? 'Yes' : 'No'
In src/kata3.numberToReversedDigits.js:
> @@ -1,5 +1,12 @@
 const numberToReversedDigits = (number) => {
 
+const tostring = number.toString();
+const split = tostring.split('');
+let map = split.map(function(result) {

map should be a constant variable here as it isn't reassigned to. I'd also use an arrow function with an implicit return:
const map = split.map(result => parseInt(result));
In src/kata4.humanCatDogYears.js:
> @@ -1,5 +1,25 @@
 const humanCatDogYears = (number) => {
 
+  let hcdy = [];
+    hcdy.push(number);
+
+    const cat = function (number) {

Really good thought process here in breaking the functionality up into different functions. I would probably not be afraid to make these more concise though:
const cat = number => (((number - 2) * 4) + 24)
In src/kata4.humanCatDogYears.js:
> +        let x = ( number - 2 ) * 4;
+          let res = x + 24;
+            return res;
+    }
+
+    hcdy.push(cat(number));
+
+    const dog = function (number) {
+        let x = ( number - 2 ) * 5;
+          let res = x + 24;
+            return res;
+    }
+
+    hcdy.push(dog(number));
+
+    return hcdy;

hcdy isn't very meaningful a name, and you probably don't need it:
return [human(number), cat(number), dog(number)];
In src/kata5.reachDestination.js:
> @@ -1,5 +1,26 @@
 const reachDestination = (distance, speed) => {
 
+  function reachDestination (distance, speed) {

Not sure why you have a function within a function here - you definitely don't need it!

In src/kata5.reachDestination.js:
> @@ -1,5 +1,26 @@
 const reachDestination = (distance, speed) => {
 
+  function reachDestination (distance, speed) {
+
+    let time = Math.floor(distance) / Math.floor(speed);
+
+    let twoDec = time.toFixed(2);
+
+    let twoDecArr = twoDec.toString().split(".");
+
+        if (twoDecArr[1] >= 0 && twoDecArr[1] <= 49) {
+            twoDecArr[1] = 5;
+        } else {
+            twoDecArr[1] = 0;
+      };
+
+    let result = twoDecArr.join('.');

Use constant variables unless it's your intention to reassign (you should avoid reassignment, but it is acceptable for mathematical operations).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
